### DISCUSSION TITLE
loop scroll in chat viewport for vue3

### DISCUSSION THREAD

**Discussion by doroved** [inokawa/virtua#465] [General]
@inokawa 
Hi, is it possible on vue3 to use virtua for a chat application with [loop scroll](https://inokawa.github.io/virtua/?path=/story/advanced-loop--default) + [restore scroll](https://inokawa.github.io/virtua/?path=/story/basics-vlist--scroll-restoration)?
Could you show a code example? Thanks

**Comment by inokawa** [comment]
Hello, scroll restoration feature is not implemented yet for Vue. #398

Probably looping scroll is possible with Vue already.

**Reply by doroved** [reply]
> Hello, scroll restoration feature is not implemented yet for Vue. #398
> 
> Probably looping scroll is possible with Vue already.

Unfortunately virtua is not a good fit for a chat application on vue as there is no reverse support.
I found this solution and it works well
https://codepen.io/zupkode/pen/MWwgLyb
I have rewritten it on vue3, but I don't have the strength to adapt it for reverse yet.

If you have a messenger https://telegram.org, you can see how scrolling is implemented there. It's perfect for chat rooms. I would like such a solution.
It uses loop scroll and it seems that to restore scroll you don't need to cache element heights, but just keep the current range(start/endIndex) and scrollTop in that range.
